scripts/cluster-upgrade.sh

```diff
@@ -342,32 +393,34 @@ check_cluster_status() {
-    
-    if [ "$DCO" != "" ];
-    then
-        log $OCM_NAME $LOG_NAME "INFO: Degraded Operators = $DCO"
-        CLUSTER_STATUS=0
-    fi
-
     # get a list of firing critical alerts in core namespaces (these would alert SREP in PD)
-    CA=$(curl -G -s -k -H "Authorization: Bearer $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring sa get-token prometheus-k8s)" --data-urlencode "query=ALERTS{alertstate=\"firing\",severity=\"critical\",namespace=~\"^openshift.*|^kube.*|^default$\"}" "https://$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring get routes prometheus-k8s -o json | jq -r .spec.host)/api/v1/query" | jq -r '.data.result[].metric.alertname' | tr '\n' ',' )
+    # NOTE exclude ClusterUpgradingSRE and DNSErrors05MinSRE
+    CA=$(curl -G -s -k -H "Authorization: Bearer $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring sa get-token prometheus-k8s)" --data-urlencode "query=ALERTS{alertstate=\"firing\",severity=\"critical\",namespace=~\"^openshift.*|^kube.*|^default$\",alertname!=\"ClusterUpgradingSRE\",alertname!=\"DNSErrors05MinSRE\"}" "https://$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-monitoring get routes prometheus-k8s -o json | jq -r .spec.host)/api/v1/query" | jq -r '.data.result[].metric.alertname' | tr '\n' ',' )
 
     if [ "$CA" != "" ];
     then
         log $OCM_NAME $LOG_NAME "WARNING: Critical Alerts = $CA"
         CLUSTER_STATUS=0
-    fi
 
-    # get a count of bad pods in core namespaces, except intsaller and pruner
-    POD_ISSUES=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get pods --all-namespaces --no-headers | grep -v -e Running -e Completed -e Terminating -e ContainerCreating -e Init -e Pending | grep -e ^default -e ^kube -e ^openshift | grep -v -e installer -e revision-pruner)
-    PPC=$(echo "$POD_ISSUES" | grep -v "^$" | wc -l)
+        # degraded operators
+        DCO=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get clusteroperator --no-headers | awk '{print $1 "," $5}' | grep -v ",False" | awk -F, '{print $1 ","}' | xargs)
+        
+        if [ "$DCO" != "" ];
+        then
+            log $OCM_NAME $LOG_NAME "INFO: Degraded Operators = $DCO"
 
-    if [ "$PPC" != "0" ];
-    then
-        log $OCM_NAME $LOG_NAME "INFO: Problem Pod Count = $PPC
-$POD_ISSUES"
-        # NOTE do not set CLUSTER_STATUS to 0 (bad), this could be a false positive
+            for CO in $DCO;
+            do
+                CO_MESSAGE=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get clusteroperator $CO -o json | jq -r '.status.extension.lastSyncError')
+
+                if [ "$CO_MESSAGE" != "" ];
+                then
+                    log $OCM_NAME $LOG_NAME "INFO: Operator $CO lastSyncError = $CO_MESSAGE"
+                fi
+            done
+
+            CLUSTER_STATUS=0
```

Comment:
The comment being removed says not to set `CLUSTER_STATUS=0` (bad) as it could be a false positive. Do we still have that concern? Is it okay to set that here?

Reply:
This can be removed, it's already set after we find a critical alert (line 403)
```suggestion

```